core/src/main/java/tc/oc/pgm/rotation/pools/MapPool.java

```diff
@@ -31,31 +27,54 @@ public abstract class MapPool implements MapOrder, Comparable<MapPool> {
   protected final boolean dynamic;
 
   MapPool(MapPoolType type, String name, MapPoolManager manager, ConfigurationSection section) {
+    this(
+        type,
+        name,
+        manager,
+        section.getBoolean("enabled"),
+        section.getInt("players"),
+        section.getBoolean("dynamic", true),
+        parseDuration(section.getString("cycle-time", "-1s")),
+        buildMapList(section.getStringList("maps"), name));
+  }
+
+  MapPool(
+      MapPoolType type,
+      String name,
+      MapPoolManager manager,
+      boolean enabled,
+      int players,
+      boolean dynamic,
+      Duration cycleTime,
+      List<MapInfo> maps) {
     this.type = type;
     this.name = name;
     this.manager = manager;
-    this.configSection = section;
-    this.enabled = section.getBoolean("enabled");
-    this.players = section.getInt("players");
-    this.dynamic = section.getBoolean("dynamic", true);
-    this.cycleTime = parseDuration(section.getString("cycle-time", "-1s"));
-
-    MapLibrary library = PGM.get().getMapLibrary();
-    List<MapInfo> mapList =
-        section.getStringList("maps").stream()
-            .map(mapName -> getMap(library, mapName))
-            .filter(Objects::nonNull)
-            .collect(Collectors.toList());
-    this.maps = Collections.unmodifiableList(mapList);
+    this.enabled = enabled;
+    this.players = players;
+    this.dynamic = dynamic;
+    this.cycleTime = cycleTime;
+    this.maps = maps;
   }
 
-  private MapInfo getMap(MapLibrary library, String mapName) {
-    @Nullable MapInfo map = library.getMap(mapName);
-    if (map != null) return map;
-    PGM.get()
-        .getLogger()
-        .warning("[MapPool] [" + name + "] " + mapName + " not found in map repo. Ignoring...");
-    return null;
+  private static List<MapInfo> buildMapList(List<String> mapNames, String poolName) {
+    List<MapInfo> mapList = Lists.newArrayList();
```

Comment:
```suggestion
    List<MapInfo> mapList = new ArrayList(mapNames.size());
```
pre-allocate for mapNames' size, avoids re-sizes as elements are added in, and we know what the final size should be from the start